packages/flutter/lib/src/rendering/custom_layout.dart

```diff
@@ -47,11 +57,20 @@ abstract class MultiChildLayoutDelegate {
         final MultiChildLayoutParentData childParentData = child.parentData;
         assert(childParentData.id != null);
         _idToChild[childParentData.id] = child;
+        assert(() {
+          _debugChildrenNeedingLayout.add(child);
+          return true;
+        });
         child = childParentData.nextSibling;
       }
       performLayout(size, constraints);
+      assert(_debugChildrenNeedingLayout.isEmpty);
```

Comment:
Yeah, printing the layout IDs of the children that didn't get laid out would be spectacular.